master/internal/db/postgres_groups.go

```diff
@@ -57,17 +65,22 @@ func (db *PgDB) AddUsersToGroup(ctx context.Context, gid int, uids ...model.User
 		})
 	}
 
-	_, err := Bun().NewInsert().Model(&groupMem).Exec(ctx)
-	return err
+	tx, err := Bun().BeginTx(ctx, &sql.TxOptions{})
```

Comment:
Nice! Thanks for looking into this and taking care of it.

We'll wanna check this error instead of overwriting it on the next line.

It's also really important with database transactions not to leave them "hanging." For example, if there was a `panic` before the `tx.Commit()`, the transaction would be left open, which can severely impact database performance or even lock a table or row (depending on DBMS, queries run in the transaction, and/or settings). Since `tx.Rollback()` is a no-op on an already committed transaction, a common idiom in Go is to `defer tx.Rollback()` immediately after successfully beginning. I think we'll want something like this:

```go
tx, err := Bun().BeginTx(ctx, &sql.TxOptions{})
if err != nil {
	return err
}
defer tx.Rollback()

res, err := tx.NewInsert().Model(&groupMem).Exec(ctx)
if err != nil {
	return checkIfFound(res, err)
}

err = tx.Commit()
return checkIfFound(res, err)
```

Reply:
Another option is to use Bun's `db.RunInTx` feature. It's 100% personal taste, and totally fine if you prefer either, but I don't like `RunInTx` as much.